group_project_v2/group_project.py

```diff
@@ -580,13 +594,17 @@ def dashboard_detail_view(self, context):
             stages.append({"id": stage.id, 'content': stage_fragment.content})
             stage_stats[stage.id] = self._get_stage_completion_details(stage, target_workgroups, target_users)
 
-        groups_data = self._build_groups_data(target_workgroups, stage_stats)
+        groups_data = self._build_groups_data(target_workgroups, stage_stats, filtered_users)
+
+        visible_groups = list(filter(attrgetter("group_visible"), groups_data))  # pylint: disable=bad-builtin
```

Comment:
Done. However I personally dislike that a new variable name was introduced: ``visible_groups = [group for group in groups_data if group["group_visible"]]``